internal/kibana/maintenance_window/response_types.go

```diff
@@ -0,0 +1,49 @@
+package maintenance_window
+
+/*
+* The types generated automatically for kibana_oapi are deeply nested a very hard to use.
+* This file defines convenience types that can be used to define these neestes objects
+* when needed.
+ */
+
+type ResponseJson struct {
+	CreatedAt string               `json:"created_at"`
+	CreatedBy *string              `json:"created_by"`
+	Enabled   bool                 `json:"enabled"`
+	Id        string               `json:"id"`
+	Schedule  ResponseJsonSchedule `json:"schedule"`
+	Scope     *ResponseJsonScope   `json:"scope,omitempty"`
+	Title     string               `json:"title"`
+}
+
+type ResponseJsonSchedule struct {
+	Custom ResponseJsonCustomSchedule `json:"custom"`
+}
+
+type ResponseJsonCustomSchedule struct {
+	Duration  string                 `json:"duration"`
+	Recurring *ResponseJsonRecurring `json:"recurring,omitempty"`
+	Start     string                 `json:"start"`
+	Timezone  *string                `json:"timezone,omitempty"`
+}
+
+type ResponseJsonRecurring struct {
+	End         *string    `json:"end,omitempty"`
+	Every       *string    `json:"every,omitempty"`
+	Occurrences *float32   `json:"occurrences,omitempty"`
```

Comment:
This is always mapped to `int32`. Is this due to how the spec is defined? I think technically there is also an `integer` type in the spec, but that would also need some upstream change in the Kibana spec.

Reply:
I was wondering the same. In Kibana, this is defined as `number`, which always leads to a `float32` generated type.

I couldn't figure out how to change the type upstream, but I thought the conversion from one to the other wouldn't be a problem.

wdyt?